Assets/SEE/XR/XRSEEActions.cs

```diff
@@ -0,0 +1,218 @@
+using SEE.Controls.Actions;
+using SEE.GO;
+using SEE.Utils;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using UnityEngine.XR.Interaction.Toolkit;
+using UnityEngine.XR.Interaction.Toolkit.Interactors;
+
+public class XRSEEActions : MonoBehaviour
+{
+    /// <summary>
+    /// The RayInteractor, which will be accessed.
+    /// </summary>
+    [SerializeField]
+    XRRayInteractor m_RayInteractor;
+    /// <summary>
+    /// The RayInteractor, which we use, to get the current position of the Laserpointer.
+    /// </summary>
+    public static XRRayInteractor RayInteractor { get; set; }
+    /// <summary>
+    /// Is true, when the user is hovering over an interactable.
+    /// </summary>
+    private bool hovering = false;
+    /// <summary>
+    /// The GameObject the user is currently hovering over.
+    /// </summary>
+    public static GameObject hoveredGameObject { get; private set; }
+    /// <summary>
+    /// The old parent of a node. We need this for the MoveAction.
+    /// </summary>
+    public static Transform oldParent { get; set; }
+    /// <summary>
+    /// The button, which is used for the primary actions.
+    /// </summary>
+    public InputActionReference inputAction;
+    /// <summary>
+    /// The button, which is used for the undo-action.
+    /// </summary>
+    public InputActionReference undo;
+    /// <summary>
+    /// The button, which is used for the redo-action.
+    /// </summary>
+    public InputActionReference redo;
+    /// <summary>
+    /// The button, which is used, to open the tooltip.
+    /// </summary>
+    public InputActionReference tooltip;
+    /// <summary>
+    /// Shows the source name of the hovered or selected object as a text label above the
+    /// object. In between that label and the game object, a connecting bar
+    /// will be shown.
+    /// </summary>
+    private ShowLabel showLabel;
+
+    // Awake is always called before any Start functions.
+    private void Awake()
+    {
+        tooltip.action.performed += Tooltip;
+        undo.action.performed += Undo;
+        redo.action.performed += Redo;
+        inputAction.action.Enable();
+        inputAction.action.performed += Action;
+        RayInteractor = m_RayInteractor;
+        Selected = false;
+    }
+    /// <summary>
+    /// This method gets called, when the user begins to hover over an interactable.
+    /// </summary>
+    /// <param name="args">Event data associated with the event when an Interactor first initiates hovering over an Interactable.</param>
+    public void OnHoverEnter(HoverEnterEventArgs args)
+    {
+        hovering = true;
+        hoveredGameObject = args.interactableObject.transform.gameObject;
+        if (GlobalActionHistory.Current() == ActionStateTypes.Move)
+        {
+            oldParent = args.interactableObject.transform.parent;
+        }
+        hoveredGameObject.transform.TryGetComponent(out showLabel);
+        showLabel?.On();
```

Comment:
The return value of `TryGetComponent` should be used here instead of a null conditional:
```suggestion
        if (hoveredGameObject.transform.TryGetComponent(out showLabel))
        {
            showLabel.On();
        }
```